dbt/adapters/bigquery/impl.py

```diff
@@ -620,12 +620,13 @@ def _partitions_match(table, conf_partition: Optional[PartitionConfig]) -> bool:
             table_field = (
                 table.time_partitioning.field.lower() if table.time_partitioning.field else None
             )
+
             table_granularity = table.partitioning_type
             conf_table_field = conf_partition.field
             return (
-                table_field == conf_table_field
+                table_field.lower() == conf_table_field.lower()
                 or (conf_partition.time_ingestion_partitioning and table_field is not None)
-            ) and table_granularity == conf_partition.granularity
+            ) and table_granularity.lower() == conf_partition.granularity.lower()
```

Comment:
Op's table actually didn't have the `granularity` key set:

```sql
-- models/foo_sans_granularity.sql
{{
  config(
    partition_by={
      "field": "updated_at",
      "data_type": "date",
    }
  )
}}

select id, updated_at, updated_at_utc from sources.raw
```

Do we need to account for that?

Reply:
> Daily partitioning is the default for all column types.

My understanding from the dbt [docs](https://docs.getdbt.com/reference/resource-configs/bigquery-configs#partitioning-by-a-date-or-timestamp) was that the default granularity is day-partitioned.

Is this where the default is set?
https://github.com/dbt-labs/dbt-bigquery/blob/main/dbt/adapters/bigquery/impl.py#L77

Reply:
yes, there is a default so it should be safe